api/retryable.go

```diff
@@ -29,6 +29,10 @@ var retryableStatuses = []int{
 	http.StatusGatewayTimeout,      // 504
 }
 
+func IsFailure(r *Response) bool {
```

Comment:
I would like more doc comments, please and thankyou 🙏

```suggestion
// IsFailure reports if the response represents a failure.
func IsFailure(r *Response) bool {
```

There's probably a better description than that, though?